beacon_node/store/src/impls/beacon_state.rs

```diff
@@ -4,6 +4,45 @@ use ssz_derive::{Decode, Encode};
 use std::convert::TryInto;
 use types::beacon_state::{CommitteeCache, CACHED_EPOCHS};
 
+pub fn store_full_state<S: Store, E: EthSpec>(
+    store: &S,
+    state_root: &Hash256,
+    state: &BeaconState<E>,
+) -> Result<(), Error> {
+    let timer = metrics::start_timer(&metrics::BEACON_STATE_WRITE_TIMES);
+
+    let bytes = StorageContainer::new(state).as_ssz_bytes();
+    let result = store.put_bytes(DBColumn::BeaconState.into(), state_root.as_bytes(), &bytes);
+
+    metrics::stop_timer(timer);
+    metrics::inc_counter(&metrics::BEACON_STATE_WRITE_COUNT);
+    metrics::inc_counter_by(&metrics::BEACON_STATE_WRITE_BYTES, bytes.len() as i64);
+
+    result
+}
+
+pub fn get_full_state<S: Store, E: EthSpec>(
+    store: &S,
+    state_root: &Hash256,
+) -> Result<Option<BeaconState<E>>, Error> {
+    let timer = metrics::start_timer(&metrics::BEACON_STATE_READ_TIMES);
+
+    let result = match store.get_bytes(DBColumn::BeaconState.into(), state_root.as_bytes())? {
+        Some(bytes) => {
+            let container = StorageContainer::from_ssz_bytes(&bytes)?;
+
+            metrics::stop_timer(timer);
+            metrics::inc_counter(&metrics::BEACON_STATE_READ_COUNT);
+            metrics::inc_counter_by(&metrics::BEACON_STATE_READ_BYTES, bytes.len() as i64);
+
+            Ok(Some(container.try_into()?))
+        }
+        None => Ok(None),
+    };
+
+    result
```

Comment:
Remove indirection.

Also check that the `timer` will stop when it gets dropped